### PR TITLE
[PBM-1142] Use UBI-based image for Go builder

### PR DESCRIPTION
**Pull request by dalbani**
Fixes https://jira.percona.com/browse/PBM-1142.

### CODE REVIEW

**Review by defbin — APPROVED** [review]

**Comment by dalbani** [comment]
Comparing my changes with https://github.com/percona/percona-backup-mongodb/pull/840, I'm wondering what this does:
https://github.com/percona/percona-backup-mongodb/blob/4f020c62284994f416730593299ef553880119af/.github/workflows/ci.yml#L38

Shouldn't it contain the Go version itself, as in `1.19`, I mean?

**Comment by defbin** [comment]
> Comparing my changes with #840, I'm wondering what this does:
> 
> https://github.com/percona/percona-backup-mongodb/blob/4f020c62284994f416730593299ef553880119af/.github/workflows/ci.yml#L38
> 
> Shouldn't it contain the Go version itself, as in `1.19`, I mean?

Best to ask @olexandr-havryliak . I guess, it is configurable with defaults to "golang:bullseye" (1.20). Used inside job scripts. But can be changed when run manually.
Do not rush to change all base images. Only `docker/*` are public. All others are for testing purposes and could be easier to use.

**Comment by defbin** [comment]
Could you please revert changes to e2e-tests images? your current changes break them

**Comment by dalbani** [comment]
> Could you please revert changes to e2e-tests images? your current changes break them

My bad, I just force pushed to revert it.

**Review by defbin — APPROVED** [review]

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,4 @@
-FROM golang:1.19-bullseye
-RUN apt-get update -y && apt-get install -y libkrb5-dev
+FROM registry.access.redhat.com/ubi8/go-toolset:1.19
```
Comment (olexandr-havryliak) [inline docker/Dockerfile:1]: missing dependency
`RUN yum -y install krb5-devel `

Comment (dalbani) [reply docker/Dockerfile:1]: But is (still) actually necessary? The application builds just fine without anyway.

Comment (olexandr-havryliak) [reply docker/Dockerfile:1]: It's necessary, but looks like ubi8/go-toolset already contains krb5-libs and krb5-devel packages

**Review by olexandr-havryliak — APPROVED** [review]

**Review by defbin — APPROVED** [review]

**Pull request merged**